schema/schema.go

```diff
@@ -118,6 +138,51 @@ type Diagnostics struct {
 	Data   Base64EncodedBytes `json:"data"`
 }
 
+// OffsetMillis is a type that serializes as uint64 in json, and represents a
+// duration in milliseconds.
+type OffsetMillis struct {
+	time.Duration
+}
+
+var (
+	_ json.Unmarshaler = (*OffsetMillis)(nil)
+	_ json.Marshaler   = (*OffsetMillis)(nil)
+)
+
+func (om *OffsetMillis) UnmarshalJSON(b []byte) error {
+	var ms uint64
+	if err := json.Unmarshal(b, &ms); err != nil {
+		return fmt.Errorf("failed to unmarshal milliseconds offset: %w", err)
+	}
+	*om = OffsetMillis{time.Duration(ms) * time.Millisecond}
+	return nil
+}
+
+func (om OffsetMillis) MarshalJSON() ([]byte, error) {
+	return json.Marshal(om.Duration.Milliseconds())
+}
```

Comment:
having all of this in schema.go is a bit distracting. maybe worth having the implementation of marshaling in a separate file.

Reply:
This was accidentally leaked from my local blockseq schema changes.

Reply:
Now rolled back.